### ISSUE TITLE
I can't count objects with YOLO when i increase the screen size

### ISSUE THREAD

**Issue by JeyMarsh** [ultralytics/ultralytics#20684]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

I'm creating a skecth to count objects with Yolo, and it works normally, but when I increase the screen resolution, it stops counting when a toy car passes through the dot region. I've tried several times to increase the dots in the region, but it doesn't work.

### Additional

import os
from ultralytics import solutions
import serial
import cv2

model = r"Repositório\runs\detect\train14\weights\best.pt"
os.environ["KMP_DUPLICATE_LIB_OK"]="TRUE"

cap = cv2.VideoCapture(0)
porta = "COM5"
velocidade = 115200
try:
    conecao = serial.Serial(porta, velocidade, timeout=0.5)
    print("Conectado com a porta", conecao.portstr)
except serial.SerialException:
    print("Não conectada")
    pass

region_points = [(20, 400), (1700, 400)]                                      # line counting
#region_points = [(0, 400), (1920, 400), (1920, 360), (0, 360)]  # rectangle region
# region_points = [(20, 400), (1080, 400), (1080, 360), (20, 360), (20, 400)]   # polygon region
cap.set(cv2.CAP_PROP_FRAME_WIDTH, 1820)
cap.set(cv2.CAP_PROP_FRAME_HEIGHT, 1080)
# Video writer
w, h, fps = (int(cap.get(x)) for x in (cv2.CAP_PROP_FRAME_WIDTH, cv2.CAP_PROP_FRAME_HEIGHT, cv2.CAP_PROP_FPS))
cap.set(cv2.CAP_PROP_FPS, 120)

# Initialize object counter object
counter = solutions.ObjectCounter(
    show=True,
    region=region_points,
    model=r"Repositório\runs\detect\train14\weights\best.pt",
    classes=[0],
     # count specific classes i.e. person and car with COCO pretrained model.
    #tracker="custom_dataset.yaml",
)

# Process video
while cap.isOpened():
    success, im0 = cap.read()

    if not success:
        print("Video frame is empty or processing is complete.")
        break
    results = counter(im0)

    in_count = results.in_count
    out_count = results.out_count
    total_present = in_count - out_count
   # boxes = results[0].boxes.xyxy.cpu().tolist()
   # clss = results[0].boxes.clss.cpu().tolist()
   # confs = results[0].boxes.conf.cpu().tolist()

    #if boxes is not None:
    #    for box, cls, conf in zip(boxes, clss, confs):
    #        color = colors(int(cls), True)

    #    cv2.rectangle(frame, (box[0], box[1]), (box[2], box[3]), color, 2)

    #    label = f'{model.names[int(cls)]} {conf:.2f}'
    #    cv2.putText(im0, label, (int(box[0]), int(box[1]) - 10), cv2.FONT_HERSHEY_SIMPLEX, 0.5, color, 2)

    # Mostra o frame processado com as caixas
   # cv2.imshow("Output", im0)
   # if cv2.waitKey(1) & 0xFF == ord('q'):  # Sai do loop se apertar a tecla 'q' for pressionada
   #     cap.release()
   #     video_writer.release()
   #     cv2.destroyAllWindows()

    # print(results)  # access the output

    print(" Total de veículos: ", total_present)
    if total_present > 0:
        conecao.write (b'1')
    else:
         conecao.write(b'0')

cap.release()
video_writer.release()
cv2.destroyAllWindows()  # destroy all opened windows
conecao.close()

**Comment by UltralyticsAssistant** [comment]
👋 Hello @JeyMarsh, thank you for reaching out and for your detailed description and code snippet! 🚀 This is an automated response to help you get the fastest and best support. An Ultralytics engineer will also review your issue soon.

We recommend visiting the [Docs](https://docs.ultralytics.com/) for more information on usage and troubleshooting. You can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) examples there.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example (MRE)](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it. This should include the smallest amount of code and data needed to reproduce the issue—this helps us help you faster!

If this is a ❓ Custom Training or Usage Question, please share as much relevant information as possible, such as sample images or videos, your dataset, and full logs. Also, check that you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community for more help and discussions:
- For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧
- For in-depth discussions, visit [Discourse](https://community.ultralytics.com/)
- Or share knowledge on our [Subreddit](https://reddit.com/r/Ultralytics)

## Upgrade

Make sure you are using the latest `ultralytics` package and all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

YOLO can be run in these up-to-date, verified environments (with all dependencies like [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/), and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

Thank you for your patience and for helping us improve Ultralytics!

**Comment by RizwanMunawar** [comment]
@JeyMarsh Hi, thanks for sharing it. You will need to adjust the region coordinates based on your video resolution to get counts properly.
